Marlin/src/HAL/HAL_AVR/HAL_AVR.h

```diff
@@ -69,6 +69,8 @@
 #define HAL_TIMER_TYPE uint16_t
```

Comment:
Also…
```cpp
typedef unsigned int timer_t;
```

Reply:
I always thought using defines for the types was for style reasons so didn't question it, would have changed it ages ago. The lpc1768 pin constants should probably also be in an enum, and use that as the pin_t type.

Reply:
We're definitely now favoring `typedef` and following the `_t` suffix convention. At least, been trying to apply that as I go along…